lib/wallet/open_channel.dart

```diff
@@ -137,40 +138,25 @@ class _OpenChannelState extends State<OpenChannel> {
               padding: const EdgeInsets.only(right: 20.0),
               child: Container(
                 alignment: Alignment.bottomRight,
-                child: ElevatedButton.icon(
-                    label: const Text('Open Channel'),
-                    icon: !submitting
-                        ? Container()
-                        : Container(
-                            width: 18,
-                            height: 18,
-                            padding: const EdgeInsets.all(2.0),
-                            child: const CircularProgressIndicator(
-                              color: Colors.white,
-                              strokeWidth: 3,
-                            ),
-                          ),
-                    onPressed: !validForm || submitting ? null : openChannel),
+                child: SubmitButton(
+                  onPressed: openChannel,
+                  label: 'Open Channel',
+                  isButtonDisabled: !validForm,
+                ),
               ))
         ],
       )),
     );
   }
 
-  void openChannel() {
-    setState(() {
-      submitting = true;
-    });
+  Future<void> openChannel() async {
     FLog.info(text: "Opening Channel with capacity " + takerChannelAmount.toString());
-    context.read<ChannelChangeNotifier>().open(takerChannelAmount).then((value) {
+    await context.read<ChannelChangeNotifier>().open(takerChannelAmount).then((value) {
```

Comment:
❓ Do we want to change this behaviour?

I know we talked about generally always awaiting, but I think this was left without an `await` on purpose by @holzeis.

Reply:
If that was *on purpose* I think it really deserves a comment :)
Happy to keep the previous behavior and I'm adding a comment. I suppose we don't `await` here because the `Open Channel` card shows the status.

Reply:
Agree on adding a comment!

Reply:
@da-kami @luckysori I guess the await is not needed anymore here, since the SubmitButton is awaiting on the complete open_channel function, so this await has simply moved 😉